google/cloud/storage/emulator/README.md

```diff
@@ -84,3 +84,40 @@ For N==1 and N==2 behave like `return-305-after-256K`, for `N>=3` ignore the
 failure instruction and return successfully. This is used to test failures during
 retry, the client cooperates by sending the retry counter in the failure
 instructions.
+
+
+## Retry Test API
+
```

Comment:
Can you write something here about the *motivation* for these new APIs and resources?  Why do we create "named" Retry Test objects instead of passing the instructions via `x-goog-emulator-instructions`?  What problem does this new approach solve?
I can make some guesses, but why guess when we can write it down so it is clear for our future selves.

Reply:
I am now really confused.  The motivation is that passing instructions in a header is hard, but passing the `x-retry-test-id: $arbitrary_string` is easy?

Reply:
The tricky part is that in several languages, configuring the client to send an extra header happens per-library call or per-client, rather than per-request sent. Therefore, it's hard to configure the client to send an extra header for the first N attempts and then stop sending it for the last request. Having the client send the same header for every request for a given test (while the emulator tracks when to stop forcing failures) is easier.

Reply:
Okay, let's say that then:

```md
## Retry Test API

The "Retry Test API" offers a mechanism to describe more complex retry scenarios while sending a single, constant header through all the HTTP requests from a test program.

Previous versions of the GCS emulator used a custom header in the RPC to control the behavior of each RPC,
for some test scenarios this required sending different header with the first retry attempt vs. subsequent attempts. Producing different headers in each attempt is not easy to implement with some client libraries.

Sending a constant header with all RPCs can be implemented across all client libraries, and to some degree decouples the test setup from the test execution.
```